pkg/ddevapp/config.go

```diff
@@ -178,6 +177,29 @@ func (c *Config) Config() error {
 	return nil
 }
 
+// Validate ensures the configuraton meets ddev's requirements.
+func (c *Config) Validate() error {
+	// validate docroot
+	fullPath := filepath.Join(c.AppRoot, c.Docroot)
+	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
+		return fmt.Errorf("no directory could be found at %s. Please enter a valid docroot in your configuration", fullPath)
```

Comment:
I'm a bit wary of doing this, as it means our detection would need to be more robust than it currently is. The big problem case I think of off-hand is that WordPress allows for a custom core directory, and a site using this structure would fail to detect as a WordPress site.